Model/Details.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Aligent Consulting
+ * Copyright (c) Aligent Consulting (https://www.aligent.com.au)
+ */
+
+declare(strict_types=1);
+
+namespace Aligent\AsyncEvents\Model;
+
+use Aligent\AsyncEvents\Api\AsyncEventRepositoryInterface;
+use Aligent\AsyncEvents\Model\ResourceModel\AsyncEventLog\Collection;
+use Aligent\AsyncEvents\Model\ResourceModel\AsyncEventLog\CollectionFactory as AsyncEventLogCollectionFactory;
+
+class Details
+{
+    /**
+     * @var array
+     */
+    private $traceCache = [];
+
+    /**
+     * @var Collection
+     */
+    private $collection;
+
+    /**
+     * @var AsyncEventRepositoryInterface
+     */
+    private $asyncEventRepository;
+
+    public function __construct(
+        AsyncEventLogCollectionFactory $collectionFactory,
+        AsyncEventRepositoryInterface  $asyncEventRepository
+    ) {
+        $this->collection = $collectionFactory->create();
+        $this->asyncEventRepository = $asyncEventRepository;
+    }
+
+    public function getDetails($uuid)
+    {
+        if (array_key_exists($uuid, $this->traceCache)) {
+            return $this->traceCache[$uuid];
+        }
+
+        $this->collection->addFilter('uuid', $uuid);
+
+        $traces = $this->collection->toArray();
+
+        if ($traces['totalRecords'] === 0) {
+            return [];
+        }
+
+        $asyncEventId = $this->collection->getFirstItem()->getData('subscription_id');
+        $asyncEvent = $this->asyncEventRepository->get($asyncEventId)->getData();
```

Comment:
Any need to check that you actually got an event id?

Reply:
Most likely not, a log entry cannot exist without it's subscription. I think the foreign key constraints have `ON DELETE CASCADE`. I think that's reasonable given that we do not allow to edit / delete a subscription once created (except disabling)

But this can itself be a discussion for sometime...